web/app/charts/MapEditor.js

```diff
@@ -0,0 +1,457 @@
+            },
+            {name: "Ablers", projection: d3.geoAlbers().scale(this.width/1.3/Math.PI)}
+        ];
+
+        let table = createTable();
+        let tbody = table.firstChild;
+        var tr = document.createElement("tr");
+        tbody.appendChild(tr);
+        parent.appendChild(table);
+        me.el = table;
+        var td;
+
+        td = document.createElement("td");
+        tr.appendChild(td);
+        let div = document.createElement("div");
+        div.className = "chart-editor-options";
+        td.appendChild(div);
+        optionsDiv = div;
+
+        td = document.createElement("td");
+        td.className = "chart-editor-preview";
+        td.style.width = "100%";
+        td.style.height = "100%";
+        tr.appendChild(td);
+        panel = createDashboardItem(td,{
+            width: "100%",
+            height: "100%",
+            title: "Untitled",
+            settings: true
+        });
+        previewArea = panel.innerDiv;
+        panel.el.className = "";
+
+
+        panel.title.onclick = function(e){
+            if (this.childNodes[0].nodeType===1) return;
+            e.stopPropagation();
+            var currText = this.innerHTML;
+            this.innerHTML = "";
+            var input = document.createElement("input");
+            input.className = "form-input";
+            input.type = "text";
+            input.value = currText;
+            input.onkeydown = function(event){
+                var key = event.keyCode;
+                if(key == 13) {
+                    panel.title.innerHTML = this.value;
+                    defaultConfig.chartTitle = this.value;
+                }
+            };
+            this.appendChild(input);
+            input.focus();
+        };
+        document.body.addEventListener('click', function(e) {
+            var input = panel.title.childNodes[0];
+            var className = e.target.className;
+            if(input.nodeType === 1 && className != "form-input") {
+                panel.title.innerHTML = input.value;
+                defaultConfig.chartTitle = input.value;
+            };
+        });
+
+        onRender(previewArea, function(){
+            initializeChartSpace();
+        });
+    };
+
+
+  //**************************************************************************
+  //** update
+  //**************************************************************************
+    this.update = function(mapConfig, inputs){
+        me.clear();
+        for (var i=0; i<inputs.length; i++){
+            var input = inputs[i];
+            if (input!=null) inputs[i] = d3.csvParse(input);
+        }
+        if(mapConfig !== null && mapConfig !== undefined){
+            Object.keys(mapConfig).forEach(val=>{
+                chartConfig[val] = mapConfig[val]? mapConfig[val]:null;
+            });
+            panel.title.innerHTML = mapConfig.chartTitle;
+            }
+        inputData = inputs;
+        createDropDown(optionsDiv);
```

Comment:
You should only create the drop downs once and show/hide what you need in the update() method

Reply:
This has been implemented by working with you a bit.